src/util/SystemInfo/SystemInfoUnix.cpp

```diff
@@ -0,0 +1,56 @@
+#if BOOST_OS_LINUX
```

Comment:
Since this file is Linux-specific with its use of `/proc/self/statm`, I suggest renaming this file with a Linux suffix instead of Unix (which itself really should be renamed to Posix everywhere IMO).
You should also add a stub (empty) implementation of `GetProcessorCount()`, `QueryRamUsage()`, `QueryProcTime()` , `QueryCoreTimes()`  for the rest of the systems (think of MacOS and the BSDs)

CMakeList should also be modified to include the platform-dependent source files, by doing so you can remove the preprocessor guards in this file and `SystemInfoWin.cpp`

@Exzap I think we need a clear naming convention for platform specific files.
How does the following sound?
`ClassName_posix.cpp` : Implementation common across posix systems.
`ClassName_win32.cpp` : Implementation specific to Windows.
`ClassName_linux.cpp` : Implementation specific to Linux.
`ClassName_macos.cpp` : Implementation specific to MacOS X.

Reply:
> Since this file is Linux-specific with its use of `/proc/self/statm`, I suggest renaming this file with a Linux suffix instead of Unix

Agreed.

> You should also add a stub (empty) implementation of `GetProcessorCount()`, `QueryRamUsage()`, `QueryProcTime()` , `QueryCoreTimes()` for the rest of the systems (think of MacOS and the BSDs)

Agreed. In `FileName_stubs.cpp`?

> CMakeList should also be modified to include the platform-dependent source files, by doing so you can remove the preprocessor guards in this file and `SystemInfoWin.cpp`

The CMakeLists contains this: `file(GLOB_RECURSE CPP_FILES *.cpp)` which makes it difficult to select files on a per-platform basis. `util/MemMapper` uses the same style of pre-processor guards, however I agree it is less than ideal. I think somebody has a PR open with some major refactors to the CMakeLists so I don't want to cause problems for them.

Reply:
Have a look at https://github.com/cemu-project/Cemu/blob/main/src/Common/CMakeLists.txt

Reply:
That doesn't use `GLOB_RECURSE`.

Reply:
Right. This kind of recursive rule is usually a bad idea lets hope this goes away soon enough.

Reply:
> > You should also add a stub (empty) implementation of `GetProcessorCount()`, `QueryRamUsage()`, `QueryProcTime()` , `QueryCoreTimes()` for the rest of the systems (think of MacOS and the BSDs)
> 
> Agreed. In `FileName_stubs.cpp`?

I think that would be ok.

Also the `GetProcessorCount()` implementation with `std::thread::hardware_concurrency()` seems portable, so you could make it common and remove the windows version.